chan.go

```diff
@@ -128,92 +157,103 @@ func (r *chanResponse) RawNext() (interface{}, error) {
 
 		return nil, io.EOF
 	case <-ctx.Done():
-		close(r.done)
 		return nil, ctx.Err()
 	}
 
 }
 
-type chanResponseEmitter struct {
-	ch   chan<- interface{}
-	wait chan struct{}
-	done <-chan struct{}
-
-	length *uint64
-	err    **cmdkit.Error
-
-	emitted bool
-}
+type chanResponseEmitter chanResponse
 
 func (re *chanResponseEmitter) Emit(v interface{}) error {
+	// Initially this library allowed commands to return errors by sending an
+	// error value along a stream. We removed that in favour of CloseWithError,
+	// so we want to make sure we catch situations where some code still uses the
+	// old error emitting semantics.
+	// Also errors may occur both as pointers and as plain values, so we need to
+	// check both.
+	if e, ok := v.(cmdkit.Error); ok {
+		v = &e
+	}
+	if e, ok := v.(*cmdkit.Error); ok {
+		log.Errorf("unexpected error value emitted: %s at\n%s", e.Error(), debug.Stack())
+	}
```

Comment:
Shouldn't this call `CloseWithError` and return?

Reply:
Nah, we don't emit errors anymore. Doing so is a programmer error, so we want them to know. But panicking because of this feels too strong, so I opted for logging an error.

Reply:
That won't cause a panic, it'll just "do the right thing" (redirect the error). As it stands, we'll end up sending the error on the value channel.

Reply:
Oh yes, I meant to return. Anyway, I change it to panic now. I hope we can then in the near future remove those panics because we are more confident we don't do that mistake anymore.